### ISSUE TITLE
Documentation for covering grammars

### ISSUE THREAD

**Issue by kylebgorman** [CUNY-CL/wikipron#490]
We have no effective documentation for the [covering grammars](https://github.com/CUNY-CL/wikipron/tree/master/data/covering_grammar) data library. 

* We should probably add a short description to the [data README](https://github.com/CUNY-CL/wikipron/blob/master/data/README.md). 
* We should give the exact instructions in the [covering grammars README](https://github.com/CUNY-CL/wikipron/blob/master/data/covering_grammar/README.md).

**Comment by neurlang** [comment]
hi what is covering grammar? I have [language to IPA longest prefix match grammar](https://github.com/neurlang/goruut/tree/master/dicts) for a number of languages:

    English
    Spanish
    German
    French
    Italian
    Arabic
    Farsi
    Luxembourgish
    Dutch
    Portuguese
    Russian
    Swedish
    Czech
    Slovak
    Romanian
    Finnish
    Isan
    Swahili
    Esperanto
    Icelandic
    Norwegian
    Jamaican
    Japanese


 It can [generate a new grammar](https://github.com/neurlang/goruut/blob/master/dicts/README.md) based on tsv files like the ones that you have

**Comment by neurlang** [comment]
Checking japanese in your dataset, I see that you have a mapping "あ"->"a̠" but also "ああ"->"a̠ː". Consider the source language word "ああ" and a target language word "a̠a̠" is this a valid covering grammar production in your case?

**Comment by kylebgorman** [comment]
> Checking japanese in your dataset, I see that you have a mapping "あ"->"a̠" but also "ああ"->"a̠ː". Consider the source language word "ああ" and a target language word "a̠a̠" is this a valid covering grammar production in your case?

Yes, it would be in that case. You can think of all the pairs in the mapping each as a substitution, then the "grammar" is simply the closure over the union of all the substitutions. 

It is a "covering" grammar because we know it's overly permissive, but it's simple enough to be specified as substitution pairs. It's useful in debugging, quality assurance, and the like.

**Comment by neurlang** [comment]
If its the case that any choice like this is valid, then it means that the idea of covering grammars is different than the one which I use (longest prefix match grammar).

In my scenario I am taking the longest match always, "ああ" and a target language word "a̠a̠" wouldn't be accepted by the grammar machine.

Since our ideas are distinct you can't simply copy paste my grammars and call it a day.
